net/Socket.hpp

```diff
@@ -1337,7 +1404,31 @@ class StreamSocket : public Socket,
             throw std::runtime_error("StreamSocket " + std::to_string(fd) +
                                      " expects a valid SocketHandler instance.");
 
-        auto socket = std::make_shared<TSocket>(std::move(hostname), fd, type, isClient, hostType, readType, creationTime);
+        bool isCounted = false;
+        if (!isClient && fd >= 0 && (type == Type::IPv4 || type == Type::IPv6))
```

Comment:
Because I don't like to leak the FD into TSocket ctor as described.
This code will close the FD and just pass -1 as FD.

The template TSocket could do something with the FD otherwise
and only later at StreamSocket ctor we would close it and use -1 -> !isOpen().

Reply:
leak the fd into the TSocket constructor ? the 'Socket' name suggests it is a C++ wrapper for the socket which is represented by the underlying fd. Many bad things could happen in theory - people could write random code without careful thought that breaks many obvious invariants in the code. The best way to avoid that is very careful thought on changes, quality code review, research and thought when writing the code, -and- having much less, much more simple code. 

What code closes the fd and passes -1 into a Socket wrapper ? why would we want to create a Socket for a -1 fd ? do we do this anywhere ? and if so - what is its lifecycle - how long does it survive ? is it long enough to be material re counting things ?